fish-rust/src/builtins/printf.rs

```diff
@@ -262,21 +262,22 @@ impl<'a, 'b> builtin_printf_state_t<'a, 'b> {
         argument: &wstr,
     ) {
         /// Printf macro helper which provides our locale.
-        macro_rules! sprintf_loc {
+        macro_rules! append_output_fmt {
             (
             $fmt:expr, // format string of type &wstr
             $($arg:expr),* // arguments
             ) => {
-                {
-                    let mut target = WString::new();
-                    sprintf_locale(
-                        &mut target,
-                        $fmt,
-                        &self.locale,
-                        &[$($arg.to_arg()),*]
-                    );
-                    target
+                // Don't output if we're done.
+                if self.early_exit {
+                    return;
```

Comment:
should probably not return from the surrounding function, only skip the rest of the macro

Reply:
Whoops, absolutely right - the danger of macros...